lib/webdrivers/common.rb

```diff
@@ -7,40 +7,52 @@ class << self
       attr_accessor :version
 
       def update
+        if correct_binary? # Already have desired or latest version
+          Webdrivers.logger.debug 'Expected webdriver version found'
+          return binary
+        end
+
+        # If site is down
         unless site_available?
```

Comment:
We've already checked for site availability, no need to make another call to get the same info. Let's memoize this method.